### PR TITLE
Processing Module v1.0

### LINKED ISSUE #53: Server Side Processing
All the processing related to server to be done.

### LINKED ISSUE #86: Add wrapper classes for Client Side Threading [ Processing Module ]
Add wrapper classes that implement java's `Runnable` interface for threading create, delete, rotate, colour change and select operations of the processing module at the client side.

https://stackoverflow.com/questions/3489543/how-to-call-a-method-with-a-separate-thread-in-java#:~:text=Create%20a%20class%20that%20implements,call%20start()%20on%20it

### LINKED ISSUE #93: Processing Module - 111701011 - ParameterizedOperations - Add Logger

### LINKED ISSUE #95: Reset Class - Clear screen

### LINKED ISSUE #100: Add User APIs (processing module)
The two classes implementing IUser APIs can be added

### LINKED ISSUE #106: Processing Side Handlers
BoardID , BoardState, Port and Object Handler Classes

### LINKED ISSUE #118: Change return type in select API
In case of select earlier we were passing ArrayList<Position> but UI wants ArrayList<Pixel> for selected object. As they are going to change the color of selected object. So while deselecting the object they would want to know the color of selected object to restore it back.

### PR DESCRIPTION
**Pull request by sakshi1603**
This PR includes complete implementation of Processing Module including server side processing (main and board server), processor class, threading, shapes, handlers, logs, exception handling. 
Closes #106 
Closes #100
Closes #95
Closes #93 
Closes #86 
Closes #53
Closes #118

### CODE REVIEW

**Comment by muhammedyaseen-git** [comment]
fine with me, awaiting your thoughts @convict-git ...

**Comment by convict-git** [comment]
> fine with me, awaiting your thoughts @convict-git ...

There's something wrong with the PR. Commits from the networking team coming up. ~~Also, 7.5K additions which don't make sense as it just should have mostly loggers and exception handling added after v0.1 (which is merged already)~~. Sorry, for this reply. 

Not sure how is it fine with you! :sweat_smile:

**Comment by sakshi1603** [comment]
> > fine with me, awaiting your thoughts @convict-git ...
> 
> There's something wrong with the PR. Commits from the networking team coming up. Also, 7.5K additions which don't make sense as it just should have mostly loggers and exception handling added after v0.1 (which is merge already).
> 
> Not sure how is it fine with you!

It does not only contain loggers and exception handling. As I mentioned above in PR description, this PR contains: Full Server Side Processing, Processor Class, Threading, Shapes, Handlers, visualiser (Test for Shapes made by @tymefighter for shapes) including logs and try catch.
Networking commits are here because there was delay in merging networking module interfaces. As most of @himanshu2047 work had dependency on networking module, he had pulled the interfaces directly from their PR but when the interfaces were merged he deleted those files and pulled from my master.

File: src/processing/utility/PriorityQueueObject.java
Comment (convict-git) [inline src/processing/utility/PriorityQueueObject.java]: Missing `}` for the method `equals`. I can resolve this issue at my end, but then it might create conflict issues as we are expecting more changes to come (in order to resolve a few bugs that you mentioned). 
So it's better the authors get it fixed.

**Review by convict-git — APPROVED** [review]

File: src/processing/server/board/ServerCommunication.java
Comment (convict-git) [inline src/processing/server/board/ServerCommunication.java]: Can we use JSON instead? It will give us more flexibility later on, to add more fields if required.

File: src/processing/server/main/Requests.java
Comment (convict-git) [inline src/processing/server/main/Requests.java]: JSON is preferred here as well.

File: src/processing/shape/LineDrawer.java
Comment (convict-git) [inline src/processing/shape/LineDrawer.java]: Seems like you missed to close the `a` tag and quotes.

File: src/processing/server/board/BoardStateRequestHandler.java
Comment (convict-git) [inline src/processing/server/board/BoardStateRequestHandler.java]: All this manipulation will become super annoying even if there's a requirement to add one extra field. Hence choosing JSON as a protocol will be much meaningful choice.

**Review by convict-git — CHANGES_REQUESTED** [review]

File: src/processing/utility/PriorityQueueObject.java
Comment (sakshi1603) [reply src/processing/utility/PriorityQueueObject.java]: Somehow you checked outdated file. I had fixed it in the last commit.

File: src/processing/server/board/ServerCommunication.java
Comment (himanshu2047) [reply src/processing/server/board/ServerCommunication.java]: Okay

File: src/processing/shape/LineDrawer.java
Comment (sakshi1603) [reply src/processing/shape/LineDrawer.java]: Fixed.

File: src/processing/utility/PriorityQueueObject.java
Comment (convict-git) [reply src/processing/utility/PriorityQueueObject.java]: Actually, that was a comment before the fix, but I forgot to submit the review request. So it shows 9 hours ago but it's from two days ago. Thanks though!

**Comment by Rahul-R5** [comment]
@sakshi1603 @himanshu2047 Does this PR concern server initialization? For server part of content module, ContentFactory can be imported and getContentServer() can be called(static method) to get the server side handler. This is subscribed to the networking module(server side) with identifier "content"

**Comment by himanshu2047** [comment]
> @sakshi1603 @himanshu2047 Does this PR concern server initialization? For server part of content module, ContentFactory can be imported and getContentServer() can be called(static method) to get the server side handler. This is subscribed to the networking module(server side) with identifier "content"

Yes both the server initialisation is there and you can provide me the startContentServer function in your factory class as discussed.

**Comment by himanshu2047** [comment]
Can be merged now, but suggest to merge it after merging content module otherwise will give build errors. @convict-git @muhammedyaseen-git

**Review by convict-git — APPROVED** [review]

**Pull request merged**
